dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -508,20 +507,29 @@ ModelMulti::evalZ(double t) {
 bool
 ModelMulti::propagateZModif() {
   vector<int> indicesDiff;
-  silentZChange_ = false;
+  silentZChange_.reset();
+  bool nonSilentZChange = false;
   for (int i = 0, iEnd = nonSilentZIndexes_.size(); i < iEnd; ++i) {
     if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
       indicesDiff.push_back(nonSilentZIndexes_[i]);
+      nonSilentZChange = true;
+    }
+  }
+  for (int i = 0, iEnd = notUsedInContinuousEqSilentZIndexes_.size(); i < iEnd; ++i) {
+    if (doubleNotEquals(zLocal_[notUsedInContinuousEqSilentZIndexes_[i]], zSave_[notUsedInContinuousEqSilentZIndexes_[i]])) {
+      indicesDiff.push_back(notUsedInContinuousEqSilentZIndexes_[i]);
+      if (!silentZChange_.getFlags(NotUsedInContinuousEquations) && !nonSilentZChange)
+        silentZChange_.setFlags(NotUsedInContinuousEquations);
```

Comment:
Wouldn't it be possible to delete the boolean nonSilentZChange and just set the NonSilentZ flag from the SilentZFlags?

Reply:
Not sure I got this one, the boolean is used to make sure that NotUsedInContinuousEquations is not raised if a non silent z has changed

Reply:
In here, what I don't like is that we never set the NonSilentZ flags but we use an additional boolean here to set the NotUsedInContinuousEquations flags whereas we could do (or at least I would like to do):

 for (int i = 0, iEnd = nonSilentZIndexes_.size(); i < iEnd; ++i) {
    if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
      indicesDiff.push_back(nonSilentZIndexes_[i]);
      **silentZChange_.setFlags(NonSilentZ);**
    }
  }
  for (int i = 0, iEnd = notUsedInContinuousEqSilentZIndexes_.size(); i < iEnd; ++i) {
    if (doubleNotEquals(zLocal_[notUsedInContinuousEqSilentZIndexes_[i]], zSave_[notUsedInContinuousEqSilentZIndexes_[i]])) {
      indicesDiff.push_back(notUsedInContinuousEqSilentZIndexes_[i]);
      if (!silentZChange_.getFlags(NotUsedInContinuousEquations) && **!silentZChange_.getFlags(NonSilentZ))**
        silentZChange_.setFlags(NotUsedInContinuousEquations);

In addition, I then find that the test in SolverImpl is a bit weak in the sense that we test NotUsedInDiscreteEquations or NotUsedInContinuousEquations but never NonSilentZ. We only use an else that is currently equal to NonSilentZ but is unclear and is error prone in the long term in my point of view:

      if (model_->getSilentZChange(NotUsedInContinuousEquations))
        state_.setFlags(SilentZNotUsedInContinuousEqChange);
      else
        state_.setFlags(ZChange);

I would prefer setting the NonSilentZ flag here and use a else if (model_->getSilentZChange(NonSilentZ)) in the SolverImpl code.

Reply:
NonSilentZ is not a flag today as it is equals to 0 (no bit is raised when we set it). Do you want me to change this?

Reply:
done